gammapy/maps/wcs.py

```diff
@@ -773,15 +773,20 @@ def solid_angle(self):
         lon = coord.lon * np.pi / 180.0
         lat = coord.lat * np.pi / 180.0
 
-        # Compute solid angle using the approximation that it's
-        # the product between angular separation of pixel corners.
+        # Compute solid angle across centres of the pixels, approximating it
+        # as a rectangle
         # First index is "y", second index is "x"
-        ylo_xlo = lon[..., :-1, :-1], lat[..., :-1, :-1]
-        ylo_xhi = lon[..., :-1, 1:], lat[..., :-1, 1:]
-        yhi_xlo = lon[..., 1:, :-1], lat[..., 1:, :-1]
+        # TODO: Calculate actual solid angle between two great circles?
```

Comment:
A priori I think that one can use great circle only for very specific `WCS` projections, e.g. for the Plate-Carrée or `CAR` projection. But for most other projections the axes do not follow great circles e.g. Hammer-Aitoff `AIT` or tangential `TAN`.

So it is possible to use exact formulae for some specific projections, but not in general.

Reply:
Okay. Shall I remove this TODO?